pico/tools/src/main/java/io/helidon/pico/tools/InterceptionPlan.java

```diff
@@ -44,6 +45,20 @@ public interface InterceptionPlan {
      */
     Set<AnnotationAndValue> serviceLevelAnnotations();
 
+    /**
+     * Returns true if the implementation has a zero/no-argument constructor.
+     *
+     * @return true if the service type being intercepted has a zero/no-argument constructor
+     */
+    boolean hasNoArgConstructor();
```

Comment:
This is irrelevant (always). I do not care if the constructor is no-arg or has 10 arguments with `@Inject`. We need to treat all of these cases equally

Reply:
https://github.com/helidon-io/helidon/issues/6542